model/map/src/main/java/org/keycloak/models/map/realm/MapRealmEntity.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.keycloak.models.map.realm;
+
+public class MapRealmEntity extends AbstractRealmEntity<String> {
```

Comment:
Do we plan to change the key to UUID in the future? Or we are ok with String for realms?

Reply:
As far as I understand it we aim to have more generic keys and let particular stores to decide what key is the best for them.

Reply:
I see. But we are using UUID across all entities, what is the reason for having String for realms?

Reply:
There are used non-uuid strings (names) as realm ids across current codebase, especially for `master` realm. https://github.com/keycloak/keycloak/blob/master/services/src/main/java/org/keycloak/services/managers/RealmManager.java#L95

There are several places which uses this behaviour: https://github.com/keycloak/keycloak/blob/master/core/src/main/java/org/keycloak/Config.java#L32

Also current installations can have non-uuid string id. 

The issue was discussed a while ago and the conclusion was (if I recall correctly) that we should stop using realmName as id, but we don't have a capacity for such change right now. We'd also need to provide migration path for current installations and take into an account old exported realm.json files.